File: 1_two_sum/1_two_sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> hashMap;

        for (int i = 0; i < nums.size(); i++)
        {
            int B = target - nums[i];
            if (hashMap.find(B) != hashMap.end())
            {
                return {hashMap[B], i};
            }
            hashMap[nums[i]] = i;
        }
        return {};
    }
};

File: 1_two_sum/1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexNum = {}

        for i, b in enumerate(nums):
            # target = a + b, -> target - b = a
            a = target - b

            if a in indexNum:
                # i --> index B
                return [indexNum[a], i]
            indexNum[b] = i
